guides/micronaut-oracle-function-http/micronaut-oracle-function-http.adoc

```diff
@@ -0,0 +1,370 @@
+= @guideTitle@
+
+@guideIntro@
+
+Authors: @authors@
+
+Micronaut Version: @micronaut@
+
+include::{commondir}/common-gettingStarted.adoc[]
+
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 11 or greater installed with `JAVA_HOME` configured appropriately
+* Docker installed
+* A paid or free trial Oracle Cloud account (create an account at https://signup.oraclecloud.com[signup.oraclecloud.com])
+* https://docs.cloud.oracle.com/en-us/iaas/Content/API/SDKDocs/cliinstall.htm[Oracle Cloud CLI] installed, with local access to Oracle Cloud configured by running `oci setup config`
+
+include::{commondir}/common-completesolution.adoc[]
+
+== Writing the App
+
+include::{commondir}/common-cli-or-launch.adoc[]
+
+[source,bash]
+----
+mn create-app --features=oracle-function,oracle-cloud-sdk example.micronaut.micronautguide --build=@build@ --lang=@lang@
+----
+
+include::{commondir}/common-build-lang-arguments.adoc[]
+
+If you use https://launch.micronaut.io[Micronaut Launch], select Micronaut Application as application type and add the `oracle-function` and `oracle-cloud-sdk` features.
+
+include::{commondir}/common-default-package.adoc[]
+
+include::{commondir}/common-annotationprocessors.adoc[]
+
+=== Dependencies
+
+Add a dependency for `oci-java-sdk-core` so we have access to classes for managing compute instances:
+
+dependency:oci-java-sdk-core[groupId=com.oracle.oci.sdk]
+
+=== InstanceData
+
+Then create an `InstanceData` DTO class to represent Compute Instance properties:
+
+source:InstanceData[]
+
+=== MicronautguideController
+
+The generated application contains a `MicronautguideController` class which is good for getting started, but we'll update it to demonstrate working with OCI SDK APIs, in this case working with Compute Instances.
+
+Replace the generated `MicronautguideController` with this:
+
+source:MicronautguideController[]
+
+<1> The controller's root uri  is `/compute`
+<2> Here we dependency-inject the SDK `ComputeClient` instance
+<3> This endpoint accepts `GET` requests and returns current instance properties for the specified instance OCID
+<4> This endpoint accepts `POST` requests and starts the specified instance if it is stopped
+<5> This endpoint accepts `POST` requests and stops the specified instance if it is running
+
+== Testing the Application
+
+NOTE: The code in this guide should work with recent versions of Micronaut 2.5+, but the tests require at least version 2.5.7
+
+We need to update the generated `MicronautguideControllerTest` to test the changes we made in `MicronautguideController`.
+
+This will be a unit test, so we'll need some mock beans to replace the beans auto-registered by the `oracle-cloud-sdk` module which make requests to Oracle Cloud. We also need a workaround for the strict classloader isolation used by https://fnproject.io/[Fn Project] (which powers Oracle Cloud Functions).
```

Comment:
Maybe don't use the word `workaround` here, it seems that is the way `Fn` is designed. Rephrase to `We also need to take into consideration the strict class loader isolation...`